### PR TITLE
ROS2 support

### PR DESCRIPTION
**Pull request by ipa-nhg**
@ipa-hsd I open the PR just to track your WIP version. Just set it as a draft for now.

### CODE REVIEW

File: rosgraph_monitor/observer.py [hunk 1]
```diff
@@ -62,12 +67,59 @@ def _stopped(self) -> bool:
...
+        diag_msg = DiagnosticArray()
+
+        diag_msg.status = [status_msg]
+
+        diag_msg.header.stamp = self._clock.now().to_msg()
+        self._pub_diag.publish(diag_msg)
+
+    def generate_diagnostics(self, *msgs) -> t.List[DiagnosticStatus]:
+        future = Future()
+        future.add_done_callback(lambda future : self.publish_diagnostics(future.result()))
```
Comment (hsd-dev) [inline rosgraph_monitor/observer.py]: can there be a timeout to receive messages? At the end of timeout a "no messages received" ERROR can be published

**Comment by hsd-dev** [comment]
created a ROS2 branch for `ros_model_parser`: https://github.com/ipa-hsd/ros_model_parser/tree/foxy-devel

**Comment by hsd-dev** [comment]
depends on https://github.com/ipa320/ros_model_parser/pull/22. Generates (but not publishes yet) diagnostics message
```
[diagnostic_msgs.msg.DiagnosticStatus(level=b'\x02', name='goal_and_stop_node', message='Missing node', hardware_id='', values=[]), 
diagnostic_msgs.msg.DiagnosticStatus(level=b'\x02', name='robot_state_publisher', message='Missing node', hardware_id='', values=[]), 
diagnostic_msgs.msg.DiagnosticStatus(level=b'\x01', name='fibonacci_action_server', message='Additional node', hardware_id='', values=[])]
```

**Comment by NadiaHG** [comment]
@ipa-hsd can this be merged? is this PR still a draft?

**Comment by hsd-dev** [comment]
The messages are populated but just the publishing is pending

**Comment by hsd-dev** [comment]
requires https://github.com/ipa320/ros_model_parser/pull/22

**Comment by hsd-dev** [comment]
Thanks for fixing the CI @ipa-nhg! Will merge the PR. Will open a new PR for further improvements.

**Pull request merged**